pkg/lib/bundle/generate.go

```diff
@@ -63,37 +82,83 @@ func GenerateFunc(directory, packageName, channels, channelDefault string, overw
 		return err
 	}
 
-	file, err := ioutil.ReadFile(filepath.Join(directory, MetadataDir, AnnotationsFile))
-	if os.IsNotExist(err) || overwrite {
-		err = WriteFile(AnnotationsFile, filepath.Join(directory, MetadataDir), content)
-		if err != nil {
-			return err
-		}
-	} else if err != nil {
+	// Push the output yaml content to the correct directory and conditionally copy the manifest dir
+	outManifestDir, outMetadataDir, err := CopyYamlOutput(content, directory, outputDir, workingDir, overwrite)
+	if err != nil {
 		return err
-	} else {
-		log.Info("An annotations.yaml already exists in directory")
-		if err = ValidateAnnotations(file, content); err != nil {
-			return err
-		}
 	}
 
 	log.Info("Building Dockerfile")
 
 	// Generate Dockerfile
-	content, err = GenerateDockerfile(mediaType, ManifestsDir, MetadataDir, packageName, channels, channelDefault)
+	content, err = GenerateDockerfile(mediaType, ManifestsDir, MetadataDir, outManifestDir, outMetadataDir, workingDir, packageName, channels, channelDefault)
 	if err != nil {
 		return err
 	}
 
-	err = WriteFile(DockerFile, directory, content)
-	if err != nil {
+	_, err = os.Stat(filepath.Join(workingDir, DockerFile))
+	if os.IsNotExist(err) || overwrite {
+		err = WriteFile(DockerFile, workingDir, content)
+		if err != nil {
+			return err
+		}
+	} else if err != nil {
 		return err
+	} else {
+		log.Info("A bundle.Dockerfile already exists in current working directory")
 	}
 
 	return nil
 }
 
+// CopyYamlOutput takes the generated annotations yaml and writes it to disk.
+// If an outputDir is specified, it will copy the input manifests
+// It returns two strings. resultMetadata is the path to the output metadata/ folder.
+// resultManifests is the path to the output manifests/ folder -- if no copy occured,
+// it just returns the input manifestDir
+func CopyYamlOutput(annotationsContent []byte, manifestDir, outputDir, workingDir string, overwrite bool) (resultManifests, resultMetadata string, err error) {
+	// First, determine the parent directory of the metadata and manifest directories
+	copyDir := ""
+
+	// If an output directory is not defined defined, generate metadata folder into the same parent dir as existing manifest dir
+	if outputDir == "" {
+		copyDir = filepath.Dir(manifestDir)
+		resultManifests = manifestDir
+	} else { // otherwise copy the manifests into $outputDir/manifests and create the annotations file in $outputDir/metadata
+		copyDir = outputDir
+
+		log.Info("Generating output manifests directory")
+
+		resultManifests = filepath.Join(copyDir, "/manifests/")
+		// copy the manifest directory into $pwd/manifests/
```

Comment:
What do we gain by copying the manifests? Could we create a symlink instead?

Additionally, the bundle specification allows the location of the manifests directory to be configurable, so `manifests` may not match that configuration.

Reply:
The goal of this function is just to scaffold. imo `manifests/` is just a nice sane default if you want to generate a bundle directory. We could create a symlink, but I think that would just add to confusion -- especially given that we have already received some feedback around complexity and different usecases for what and how people want to use this.